### PR TITLE
yarnJuan carlos arias

### PR DESCRIPTION
**Pull request by striwensko**

### CODE REVIEW

**Comment by striwensko** [comment]
@charliesbox @renemichel i get some error while running the test on the system you have placed, however when i run as local i dont get any error, is it possible something is set up incorrectly i even make a simple test in which i created an app component no store, router,redux or anything, just placing a message and still got error, from travis cli, any clue on how to fix? thanks

File: src/actions/favorites.js [hunk 1]
```diff
@@ -0,0 +1,16 @@
+const FAVORITE_TOGGLE = 'favorite_toggle';
+const FAVORITE_FILTER = 'favorite_filter';
+/* Toggle Favorite */
+const toggleFavorite = (payload) => ({ type: FAVORITE_TOGGLE, payload });
+const filterFavorite = (filter) => ({ type: FAVORITE_FILTER, filter});
```
Comment (Robfz) [inline src/actions/favorites.js]: Action creators should wrap the payload inside an object:

```
const toggleFavorite = (gif) => (
  type: FAVORITE_TOGGLE,
  payload: { filter },
);
```
This is in order to have a standard way of transporting payloads, and
comply with the Flux standard.

File: src/actions/trending.js [hunk 2]
```diff
@@ -0,0 +1,32 @@
+const TRENDING_REQUEST = 'trending_request';
+const TRENDING_FAILED = 'trending_failed';
+const TRENDING_COMPLETED = 'trending_completed'
+
+const getTrendingRequested = () => ({
+    type: TRENDING_REQUEST,
+});
```
Comment (Robfz) [inline src/actions/trending.js]: Actions should always have payloads (even if they are empty objects).

File: src/components/SearchResults/SearchResults.js [hunk 3]
```diff
@@ -0,0 +1,35 @@
...
+  }
+};
+
+const mapStateToProps = (state) => {
+  return {
+    search: state.search
+  };
+};
+
+export default connect(mapStateToProps, null)(SearchResultsComponent);
```
Comment (Robfz) [inline src/components/SearchResults/SearchResults.js]: You can skip the `mapDispatchToProps` argument here:

`export default connect(mapStateToProps)(SearchResultsComponent);`

File: src/reducers/giphy.test.js [hunk 4]
```diff
@@ -0,0 +1,79 @@
+/*import giphy from '../actions/giphy';
+
```
Comment (Robfz) [inline src/reducers/giphy.test.js]: Try to remove commented code when committing.

File: src/reducers/index.js [hunk 5]
```diff
@@ -0,0 +1,15 @@
+import { combineReducers } from 'redux';
+
+import giphy from './giphy';
+import favorites from './favorites'
+import trending from './trending';
+import search from './search';
+import menu from './menu';
+
+export default combineReducers({
```
Comment (Robfz) [inline src/reducers/index.js]: 👍

File: src/App.css [hunk 6]
```diff
@@ -1,28 +0,0 @@
-.App {
```
Comment (charliesbox) [inline src/App.css]: +1 removing this css file

File: src/components/FavoriteList/FavoriteList.js [hunk 7]
```diff
@@ -0,0 +1,71 @@
...
+import { Title, Content, FavoriteList, FilterBar } from "./FavoriteList.styled";
+import favorites from '../../actions/favorites';
+import GifThumb from "../GifThumb";
+import SVG from "../SVG";
+
+export class FavoriteListComponent extends React.Component {
+	constructor(props) {
+		super(props);
+		this.input = null;
+		this.filter = this.filter.bind(this);
```
Comment (charliesbox) [inline src/components/FavoriteList/FavoriteList.js]: +1 binding this in the constructor

another solution could be using arrow functions in the method

File: src/components/FavoriteList/FavoriteList.js [hunk 8]
```diff
@@ -0,0 +1,71 @@
...
+		{
+			this.filter();
+		}
+	}
+	componentWillUnmount()
+	{
+		this.props.filterFavorite('');
+	}
+	render() {
+		var self = this;
```
Comment (charliesbox) [inline src/components/FavoriteList/FavoriteList.js]: not regarding react, but avoid using `var`, as `let` and `const` are scoped by block and `var` is scoped to function (so, buggier)

File: src/components/Menu/Menu.js [hunk 9]
```diff
@@ -0,0 +1,106 @@
...
+		this.timeLine.pause();
+	}
+	componentWillReceiveProps(props)
+	{
+		this.timeLine.direction = (props.menu.show ? 1 : -1);
+		this.timeLine.play();
+	}
+	refresh()
+	{
+		this.forceUpdate();
```
Comment (charliesbox) [inline src/components/Menu/Menu.js]: this is sort of `black magic` from react. use it with caution.

File: src/global.js [hunk 10]
```diff
@@ -0,0 +1,34 @@
+import {injectGlobal} from 'styled-components';
+injectGlobal`
```
Comment (charliesbox) [inline src/global.js]: +1 inject global

File: src/sagas/index.js [hunk 11]
```diff
@@ -0,0 +1,29 @@
+import { all, takeEvery, call, put } from 'redux-saga/effects';
```
Comment (renemichel) [inline src/sagas/index.js]: For an implementation like this it's ok to have it all in one file. But as this grow bigger you are going to have to separate into individual files. Also please note that sagas are not only for API fetching, some features like favorite toggling could be done with sagas.

**Pull request closed without merging**